_posts/2024-08-27-cheri-myths-safe-languages.markdown

```diff
@@ -0,0 +1,156 @@
+---
+layout: post
+title:  "CHERI Myths: I don't need CHERI if I have safe languages"
+date:   2024-08-27
+categories: cheri myths
+author: David Chisnall
+---
+
+There's a recurring myth that CHERI and safe languages are solving the same problems and that, if you have one, you don't need the other.
+When I joined the CHERI project in 2012, my primary motivation was producing hardware that enabled safe interoperability between languages, so I never felt that safe languages and CHERI were in tension.
+Quite the reverse: My work on CHERI was motivated by a desire to enable safe language adoption.
```

Comment:
I feel like this buries the lede under anecdote that might not hook the reader.  Maybe...

```suggestion
Depending on one's preferred side, the arguments go something like:
* CHERI enables converting confidentiality- or integrity-violating memory safety problems to deterministic crashes, what more could one want, or
* Safe languages completely rule out memory safety bugs for code written in any one of those languages, so what's left?
To which the answers are:
* Availability matters, too!  Being able to prove statically or recover dynamically from failures requires a significantly revised _programming model_, more than "just add CHERI flavoring".
* There are _multiple_ safe languages, and they may need to interoperate within one program!  Each safe language should view, at least, the _unsafe_ runtime of the others (as well as any "native code") with skepticism.
```